threescale_api/resources.py

```diff
@@ -910,8 +910,39 @@ def _extract_resource(self, response, collection) -> Union[List, Dict]:
             extracted = extracted.get(self._entity_collection)
         return extracted
 
+    def select_by(self, **params):
+        """Select by params - logical and
+        Usage example: select_by(role='admin')
+        Filtering by some params can be done
+        on the backend.
+         -------------------------------------
+        | Endpoint         | Filters          |
+         -------------------------------------
+        | Sections #index  | parent_id        |
+        | Files #index     | section_id       |
+        | Templates #index | type, section_id |
+         -------------------------------------
+        Args:
+            **params: params used for selection
+        Returns: List of resources
+        """
+        log.debug("[SELECT] By params: %s", params)
+
+        filters = {fil: params.pop(fil) for fil in self.__class__.FILTERS if fil in params}
```

Comment:
I've expanded pydoc string for `select_by` to better explain `FILTERS` purpose. Removing `FILTERS` from `params` is intentional because they are used directly in query(filtering is done on backend) and they should not be part of function `predicate` to filter result of `select` function.